src/code.c

```diff
@@ -3230,10 +3230,8 @@ static Int InitKernel (
     MakeBagTypePublic(T_BODY);
 #endif
 
-#if !defined(HPCGAP)
```

Comment:
Here and in the following three files: Why can we drop `#if !defined(HPCGAP)`? If this is obvious, please ignore my question, I don't understand much of HPCGAP. Ping @wilfwilson to have a look at this.

Reply:
`InitGlobalBag` simply does nothing in HPC-GAP; or more specifically: with the Boehm garbage collector. My guess is that at some stage of the HPC-GAP development, there was a move to disable all `InitGlobalBag` calls for HPC-GAP. But as you can see already a few lines below this block, that never went far. (@rbehrends or @stevelinton may know the background better).

In any case, when we merged HPC-GAP back into GAP, instead of removing those calls, we put them into `#ifdef HPCGAP`. But this serves no purpose in the end, it just makes the code confusing ("huh, why is this `InitGlobalBag` disabled for HPCGAP? And why isn't the next?")

Reply:
Yes, this should be removed, I think, especially as we want it to work with the Julia GC, which uses the same roots as GASMAN. I think it may actually be a good idea to have the Boehm GC at some time declare its global roots explicitly (it normally defaults to scanning the entire data segment, which is both inefficient and may hide bugs). That's not a pressing issue, though, just something to keep in mind.

Reply:
Note that for HPC-GAP, these calls could not stay in `InitKernel`, as `InitKernel` is only called once, from the main thread. So we'd need something like a "per-thread-InitKernel". See also #2566